Environment for GitHub Web Hooks processing

indygreg opened the pull request:
This commit implements an environment for GitHub Web Hooks processing.

Details of how everything works are in the README.

I attempted to import existing state from the moz-devservices
account. However, not all items could be imported. I'm tempted
to delete everything from the AWS account and use Terraform
to define everything from a clean slate to make sure it works.

We should probably coordinate on this pull request landing, as I'm
not sure if any of this works. `terraform plan` seems reasonable,
however.

klibby commented on github-webooks/s3.tf:
```diff
@@ -0,0 +1,4 @@
+resource "aws_s3_bucket" "webhooks_bucket" {
+  bucket = "moz-github-webhooks"
+  acl = "private"
+}
```
One of the few resources in this plan that supports tags! I have a (draft, but no changes suggested yet) tagging standard at https://mana.mozilla.org/wiki/display/DEVSERVICES/Resource+Tagging; some of that feels a lot for an S3 bucket, so I'm happy for feedback.

klibby commented on github-webooks/iam-roles.tf:
```diff
@@ -0,0 +1,168 @@
+}
+
+resource "aws_iam_role_policy" "lambda_github_webhooks_receive_policy" {
+  name = "lambda-github-webhooks-receive"
```
Nit: we've been using 4-space indentation everywhere; I don't hugely mind 2 or 4, but both in the same file irks me.

klibby reviewed (changes requested):
Nicely done! 

Is there any chance that we'd want to have a non-production version of this? It's not obvious is this account repo, but @dividehex and I have been using "servicename-environment" as directory names for organization. It gets used as $tfenv in init.sh to separate remote state storage, so that you never mix dev and prod, for example. It's not a requirement, just something to think about early on.

indygreg committed:
Environment for GitHub Web Hooks processing

This commit implements an environment for GitHub Web Hooks processing.

Details of how everything works are in the README.

I attempted to import existing state from the moz-devservices
account. However, not all items could be imported. I'm tempted
to delete everything from the AWS account and use Terraform
to define everything from a clean slate to make sure it works.

We should probably coordinate on this pull request landing, as I'm
not sure if any of this works. `terraform plan` seems reasonable,
however.

indygreg commented:
I just pushed updates to this but it doesn't look like GitHub's UI updated :/

I successfully ran `terraform apply` against the moz-devservices account, so this code seems to be working!

To answer @fubar I don't believe we'll ever have a non-prod deployment of this code. Although we may want to spin up a test cluster in the future. YAGNI?

klibby commented:
rgr. if we do need a test cluster then we can deal with it then.
